### DISCUSSION TITLE
How to log JSON format in middleware?

### DISCUSSION THREAD

**Discussion by satriahu** [sidekiq/sidekiq#5038] [Q&A] [answered]
so I've been developing a middleware to log jobs data (TID, retry count, etc).
but the logger (or `Sidekiq.logger`) only accepts strings, as far as I know.

What should I do to log the data?
Any suggestions?

my raw codes may look like this (I treat it like Rails logger) : 
<img width="583" alt="Screen Shot 2021-10-29 at 19 09 07" src="https://user-images.githubusercontent.com/81661682/139431881-6675d1af-8aa9-46a3-ae1f-183a246943b2.png">

**Comment by mperham** [comment, marked as answer]
you have to add key/values to the job context. Keep reading the source.
